src/index.ml

```diff
@@ -157,17 +157,17 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
 
   let iter_io ?min ?max f io = iter_io_off ?min ?max (fun _ e -> f e) io
 
-  type window = { buf: bytes; off: int64 }
+  type window = { buf : bytes; off : int64 }
 
   let get_entry io ~window off =
     match window with
     | None ->
-      let buf = Bytes.create entry_size in
-      let _ = IO.read io ~off buf in
-      decode_entry buf 0
+        let buf = Bytes.create entry_size in
+        let _ = IO.read io ~off buf in
```

Comment:
If `IO.read` doesn't completely fill `buf` here (hopefully `~off` is an offset into `io`?), the uninitialized memory from `Bytes.create` will be in `buf` when `decode_entry` is called.
Same for the other cases where the result is ignored.